src/app/screens/Calculator/index.js

```diff
@@ -0,0 +1,50 @@
+import React, {useState} from 'react';
+import {View, TextInput} from 'react-native';
+import {KeyboardAwareScrollView} from 'react-native-keyboard-aware-scroll-view';
+
+import styles from './styles';
+import Button from './components/Button';
+import { retrieveButtons } from './utils';
+
+const Calculator = () => {
+  //Variables y funcionesa
+  const [display, setDisplay] = useState('');
+
+  const leerPresionado = target => {
+    GET_BUTTONS.find(button => button.label === target).action();
+  };
+
+  let GET_BUTTONS = retrieveButtons(display, setDisplay);
+
+  //Fin de variables y funciones
+  return (
+    <KeyboardAwareScrollView style={styles.contain}>
+      <View style={styles.containerCalculadora}>
+        <View style={styles.boxPantalla}>
+          <View style={styles.pantalla}>
+            <TextInput
+              onKeyPress={({nativeEvent}) => {
+                (nativeEvent.key.match(/[0123456789]/) ||
+                  nativeEvent.key.match(/[+-/=%*,]/)) != null
+                  ? leerPresionado(nativeEvent.key)
+                  : nativeEvent.key == 'Backspace'
+                  ? GET_BUTTONS.find(button => button.label == '<').action()
+                  : console.warn(
+                      'Introduzca números o caracteres de una calculadora.',
+                    );
+              }}
```

Comment:
This is very messy, extract all this logic to a function.

Reply:
Put the function in utils.